### PR TITLE
added segmentio telemetry

### PR DESCRIPTION
**Pull request by jasmingacic**
Added data anonymous data collection.  Basically it collects information about the host in form like this:
```
{
  "anonymousId": "a4652358effb311a074bf84d2aed5a7d270dee858bff10e847df2a9ea132bb38",
  "context": {
    "library": {
      "name": "analytics-go",
      "version": "3.0.0"
    }
  },
  "event": "testkube-heartbeat",
  "integrations": {},
  "messageId": "2021-11-04 19:54:40.029549 +0100 CET m=+0.148209228",
  "originalTimestamp": "2021-11-04T19:54:40.029571+01:00",
  "receivedAt": "2021-11-04T18:54:41.004Z",
  "sentAt": "2021-11-04T18:54:40.029Z",
  "timestamp": "2021-11-04T18:54:41.004Z",
  "type": "track"
}
```

If we are going to proceed with this we need to make sure that LD_FLAG is passed 
```
LD_FLAGS += -X github.com/kubeshop/testkube/pkg/telemetry.telemetryToken=$(TELEMETRY_TOKEN)
```

Also `TESTKUBE_TELEMETRY_DISABLE` can be passed to the API server to disable data collection.

I've created integration with my personal account for testing purposes which I can share in slack

Signed-off-by: Jasmin Gacic <jasmin.gacic@gmail.com>

### CODE REVIEW

**Comment by exu** [comment]
Ldflags are added only in makefile 
I think we have also some docker bake commands in GitHub action so probably it'll need to added there also ?

**Comment by exu** [comment]
Do we want to embed key in binary ? I think it'll be quite easy to retrieve

**Comment by exu** [comment]
Missing docs describing what kind of data are we sending and how to disable it on demand

**Comment by jasmingacic** [comment]
Let's find the best options for deployment and I will add docs. Open questions are how the users will be able to opt out of telemetry.

Embedding the key into the app is the easiest way plus even if we exposed it no harm done. Also let's see what @fog1985 suggests on how to add that to GH actions

**Review by exu — CHANGES_REQUESTED** [review]

File: cmd/api-server/main.go [hunk 1]
```diff
@@ -27,6 +28,10 @@ func init() {
 
 func main() {
 
+	if _, telemetryNotEnabled := os.LookupEnv("TESTKUBE_TELEMETRY_DISABLED"); !telemetryNotEnabled {
```
[the remark this answers is not in the document]
Comment (jasmingacic) [reply cmd/api-server/main.go:31]: by default it is enabled, unless you specify otherwise
we want to get as much data as possible

**Review by exu — CHANGES_REQUESTED** [review]
still missing docs

**Comment by fog1985** [comment]
> Let's find the best options for deployment and I will add docs. Open questions are how the users will be able to opt out of telemetry.
> 
> Embedding the key into the app is the easiest way plus even if we exposed it no harm done. Also let's see what @fog1985 suggests on how to add that to GH actions

Hey @jasmingacic ,
Not sure what is meant to be done on GH's side. Please clarify so that I could respond. properly.

**Comment by jasmingacic** [comment]
@fog1985 so when we bake our release we should inject [LD_FLAG](https://github.com/kubeshop/testkube/blob/telemetry/Makefile#L10) which is passed to build [process](https://github.com/kubeshop/testkube/blob/telemetry/Makefile#L25) the value of LD_FLAG should be stored as a secret somewhere in GH actions.

File: docs/telemetry.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Telemetry
+
+To improve the end-user experience TestKube collects anonymous data about usage and sends it to us.
+
+Telemetry collects and scrambles infromation about the host when the API server is bootstrapped for the first time.
```
Comment (olensmar) [inline docs/telemetry.md:5]: ", allowing us to ... " ?

Comment (olensmar) [inline docs/telemetry.md:5]: typo - infromation

**Review by exu — APPROVED** [review]

**Pull request merged**